sbin/upload-salt-reports

```diff
@@ -108,6 +108,17 @@ def upload(jobs):
         if job['job']['result'] == {}:
             continue
 
+        # Convert duration to seconds to conform with Puppet run
+        # duration, and in turn, the run time on Foreman reports
+
+        for node, states in job['job']['result'].iteritems():
+            if isinstance(states, dict):
+                for state, params in states.iteritems():
+                    for key, value in params.iteritems():
+                        if key == 'duration' and isinstance(value, float):
+                            value = value / 1000
```

Comment:
Some versions of Salt actually return this as a string :-( E.g,.  in the format "500 ms" -> we handle that in the report processor in the foreman plugin